apps/condo/pages/auth/index.tsx

```diff
@@ -0,0 +1,111 @@
+            <Row justify='center' gutter={[0, 40]}>
+                <ResponsiveCol desktopWidth={350}>
+                    <Row gutter={[0, 40]}>
+                        <Col span={24}>
+                            <Typography.Title level={2}>
+                                {WelcomeMessage}
+                            </Typography.Title>
+                        </Col>
+                        <Col span={24}>
+                            <Row gutter={[0, 24]}>
+                                <Col span={24}>
+                                    <AuthButton type='secondary' block onClick={async () => {
+                                        setCookie(AUTH_FLOW_USER_TYPE_COOKIE_NAME, 'resident', { maxAge: COOKIE_MAX_AGE_IN_SEC })
+                                        await router.replace('/auth/resident')
+                                    }}>
+                                        {IAmResidentMessage}
+                                    </AuthButton>
+                                </Col>
+                                <Col span={24}>
+                                    <AuthButton type='secondary' block onClick={async () => {
+                                        setCookie(AUTH_FLOW_USER_TYPE_COOKIE_NAME, 'staff', { maxAge: COOKIE_MAX_AGE_IN_SEC })
+                                        await router.replace(isValidNextUrl ? `/auth/register?next=${encodeURIComponent(next)}` : '/auth/register')
+                                    }}>
+                                        {IAmOrganizationRepresentativeMessage}
+                                    </AuthButton>
+                                </Col>
+                            </Row>
+                        </Col>
+                    </Row>
+                </ResponsiveCol>
+            </Row>
+        </>
+    )
+}
+
+AuthPage.container = AuthLayout
+AuthPage.skipUserPrefetch = true
+AuthPage.getPrefetchedData = async ({ context }) => {
+    const hasResidentApp = residentAppInfo?.mobile?.help && residentAppInfo?.mobile?.download
+
+    const userType = getCookie(AUTH_FLOW_USER_TYPE_COOKIE_NAME, { req: context?.req, res: context?.res })
+    const wasAuthenticated = getCookie(WAS_AUTHENTICATED_COOKIE_NAME, { req: context?.req, res: context?.res })
+
+    const next = context?.query?.next
+    const isValidNext = !Array.isArray(next) && isSafeUrl(next)
+
+    let nextUrl = null
+    if (wasAuthenticated) {
+        nextUrl = isValidNext ? `/auth/signin?next=${encodeURIComponent(next)}` : '/auth/signin'
+    } else if (userType === 'staff') {
+        nextUrl = isValidNext ? `/auth/register?next=${encodeURIComponent(next)}&step=inputPhone` : '/auth/register?step=inputPhone'
+    } else if (!hasResidentApp) {
+        nextUrl = isValidNext ? `/auth/register?next=${encodeURIComponent(next)}&step=inputPhone` : '/auth/register?step=inputPhone'
+    } else if (userType === 'resident') {
+        nextUrl = '/auth/resident'
```

Comment:
Do we really need to save user type flow? Looks complicated. Why not just redirect him to /signin or /register page?